src/linker/Linker.Steps/LinkAttributesParser.cs

```diff
@@ -0,0 +1,371 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Xml.XPath;
+using Mono.Cecil;
+
+namespace Mono.Linker.Steps
+{
+	public class LinkAttributesParser : ProcessLinkerXmlStepBase
+	{
+		AttributeInfo _xmlInfo;
+
+		public LinkAttributesParser (XPathDocument document, string xmlDocumentLocation)
+			: base (document, xmlDocumentLocation)
+		{
+		}
+
+		public LinkAttributesParser (XPathDocument document, EmbeddedResource resource, AssemblyDefinition resourceAssembly, string xmlDocumentLocation = "<unspecified>")
+			: base (document, resource, resourceAssembly, xmlDocumentLocation)
+		{
+		}
+
+		public AttributeInfo Parse (LinkContext context)
+		{
+			_xmlInfo = new AttributeInfo ();
+			Process (context);
+			return _xmlInfo;
+		}
+
+		public void Parse (LinkContext context, AttributeInfo xmlInfo)
+		{
+			_xmlInfo = xmlInfo;
+			Process (context);
+		}
```

Comment:
Same comment as for the substitution - I would prefer just one overload and context in .ctor